pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -904,7 +909,14 @@ func (c *Controller) WorkloadInstanceHandler(si *model.WorkloadInstance, event m
 	case model.EventDelete:
 		delete(c.workloadInstancesByIP, si.Endpoint.Address)
 	default: // add or update
+		// Check to see if the workload entry changed. If it did, clear the old entry
+		k := si.Name + "~" + si.Namespace
```

Comment:
Not related to this but - We are using "~" separator at multiple places in the code, May be we should move that to some constant like `ISTIO_SEPARATOR`  :-) or some thing and use it every where so that in future for some odd reason if we have to change, we can change easilt